### ISSUE TITLE
Request: Add Issue templates

### ISSUE THREAD

**Issue by bkaestner** [SystemCrafters/crafted-emacs#46]
Given the influx of issues lately, I believe that proper GitHub issue templates provide a more coherent experience for both the issue handlers as well as the issue writers.

**Comment by bkaestner** [comment]
Some ideas for issue templates

- Feature Request
- Package Request
- Bug Report

Some ideas for pull-request templates:

- Feature Implementation / Package addition
- Bug Fix
- Configuration Change

We might also want to add automatic labels to those templates, but I'm not sure about GitHub's feature set. I know that GitLab supports those via Quick Actions.

**Comment by daviwil** [comment]
Those sound like a great starting point!  It looks like GitHub recently added the ability to automatically add specified labels to issue/PR templates (among other things):

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

**Comment by jvdydev** [comment]
I played around with Issue Forms a bit in a repo, they do look interesting.
It's basically just YAML (for anyone who hasn't looked into it), not overly complicated (although I wish one didn't have to push to the repo to get info whether or not it's in the correct format...).

Sample I put together, obviously contents/layout would need to be discussed but I think this might be an interesting option.

![ce-bug-report-form-sample](https://github.com/SystemCrafters/crafted-emacs/assets/62522175/7b17f29f-e259-4338-9abc-4818e2725fd2)

We could/should always offer a separate "Report an issue" that is just markdown (like unformatted issues) as a separate template.

**Comment by jvdydev** [comment]
Also made a quick one for package requests.
Just samples, of course.

![ce-package-request-form-sample](https://github.com/SystemCrafters/crafted-emacs/assets/62522175/dfd6dd4b-de16-49e7-a736-4e511658722e)

**Comment by sthesing** [comment]
These look great, thanks! 
Regarding the discussion about contents: 
For issues that report an error, it would be probably good to ask explicitly for the error message (though that probably goes into "what happened"), a backtrace and a bit of code to reproduce the error. 
Of course not every user reporting an issue will be able to provide that, but asking for it and maybe providing hints on how to generate that information would be helpful, I think.

**Comment by jvdydev** [comment]
Might be as simple as changing the "subtitle" of the "What happened?" towards

```md
Please describe the bug you encountered. 
If you can, please provide a step-by-step way to reproduce the bug and/or backtrace(s) of the bug.
```

**Comment by jeffbowman** [comment]
For the backtrace, it would probably be a good idea to mention `(debugon-on-error)`.  I agree with getting a list of steps to reproduce the problem.  Also a minimal configuration would be very helpful as well, so a link to a gist or a repo where such a file exists (or even their own configuration!) .  I usually spin up a VM and a clean-room environment where I can attempt to reproduce problems in bug reports, so every bit of information is useful.

Other notes: we only support released versions of Emacs.  I don't see the point in debugging Crafted Emacs for development versions of Emacs.  I don't mind trying to be helpful, but I'm not likely to change the codebase for something in a future version of Emacs since the API in that future version may not work in the released versions.  Had a couple of issues in the past where someone said "This doesn't work on Android in Emacs 30" before even 29 was release (!!). 

It might be a good idea to get the OS and level they are on, ie a drop down for Linux, Mac, Windows and a text box for them to type something like "RHEL" or "Ventura" or "11" or whatever.

**Comment by bkaestner** [comment]
> Of course not every user reporting an issue will be able to provide that, but asking for it and maybe providing hints on how to generate that information would be helpful, I think.

A command that would bundle all those informations in a single Markdown could be helpful:

    M-x crafted-emacs-issue-information-bundle

**Comment by jvdydev** [comment]
I think it would be nice to have a function displaying information about the current Emacs session, so it can be added into the form.

Although I'm not sure it's a good idea to have a "bundle issue info".
If a user has problems to the point where they cannot bootstrap `Crafted Emacs`, they wouldn't be able to get to that function.

**Comment by jvdydev** [comment]
![ce-bug-report-sample-v2](https://github.com/SystemCrafters/crafted-emacs/assets/62522175/17155883-477f-43b9-84ac-fab830c494f6)

OS options:
- Linux/BSD
- Windows
- Mac
- Android
- other

Emacs Version options:
- 27
- 28
- 29

I added more descriptions as well as an input field for additional OS information.

**Comment by jvdydev** [comment]
Also modified the package request form a little.

![ce-pkg-request-form-sample-v2](https://github.com/SystemCrafters/crafted-emacs/assets/62522175/b4d3ecec-566b-4d29-9b57-4d9f70656695)

Also since I never listed out the options ...
Minimum Emacs Versions:
- less than 27
- 27
- 28
- 29

Package Archive:
- GNU ELPA
- NonGNU ELPA
- MELPA Stable
- MELPA

Added additional requirements text field (e.g. if there are requirements to have Emacs built a certain way etc.).
I still need to update some of the descriptions.

**Comment by sthesing** [comment]
This looks really good, thanks for preparing this!

**Comment by jvdydev** [comment]
well, the rendering is GitHub, my local test renderer looks ... let's leave it at "not as good" :laughing:.

Unless anyone has more suggestions, I'll make a PR towards the end of the week.

**Comment by jeffbowman** [comment]
No additional suggestions from me, this looks good!!

**Comment by jeffbowman** [comment]
closing this since we merged #374 already.
